control/timeresp.py

```diff
@@ -1358,6 +1358,12 @@ def step_response(sys, T=None, X0=0., input=None, output=None, T_num=None,
     xout = np.empty((sys.nstates, ninputs, np.asarray(T).size))
     uout = np.empty((ninputs, ninputs, np.asarray(T).size))
 
+    # Get labels for the inputs/outputs (prior to mimo -> simo conversion)
+    input_labels = sys.input_labels if input is None \
+        else [sys.input_labels[input]]
+    output_labels = sys.output_labels if output is None \
+        else [sys.output_labels[output]]
+
```

Comment:
I was thinking about it. Should we move it to here, or should we make the mimo/siso/simo conversions to also propagate the labels (and name)?

Reply:
Either way works, but because `_get_ss_simo` is an internal function, it is fine for it to ignore labels.

Reply:
take a look at [this commit](https://github.com/python-control/python-control/commit/a711f5530eec8292729b565e333f4372887b9686)

Reply:
It's your PR, so feel free to revert the moves that I made and put in your proposed commit.  I was just trying to get something that passed the unit tests, since they were failing.

Reply:
Also: I think it should work to do `sys.output_labels[output]` instead of `[sys.output_labels[output]]` (eg, remove the outer list).  The code underneath is smart enough to deal with getting passed a string for a single input/output.

Reply:
> Also: I think it should work to do `sys.output_labels[output]` instead of `[sys.output_labels[output]]` (eg, remove the outer list). The code underneath is smart enough to deal with getting passed a string for a single input/output.

The _process_labels had a small bug, now it works with plain str instead of [str]